utreexo/rev.go

```diff
@@ -0,0 +1,352 @@
+package utreexo
+
+import (
+	"encoding/binary"
+	"errors"
+	"io"
+	"math"
+	"math/big"
+	"os"
+)
+
+// RevReader ...
+type RevReader struct {
+	Records []*BlockRecord
+}
+
+// BlockRecord ...
+type BlockRecord struct {
+	Magic uint32
+	Size  uint32
+	Data  []byte
+	Block *CBlockUndo
+	Hash  []byte
+}
+
+// CBlockUndo ...
+type CBlockUndo struct {
+	Txs []*CTxUndo
+}
+
+// CTxUndo ...
+type CTxUndo struct {
+	Ins []*CTxInUndo
+}
+
+// CTxInUndo ...
+type CTxInUndo struct {
+	Height uint64
+	Script []byte
+	Amount uint64
+}
+
+// OpenRevReader ...
+func OpenRevReader(name string) (*RevReader, error) {
+	f, err := os.Open(name)
+	if err != nil {
+		return nil, err
+	}
+	fi, err := f.Stat()
+	if err != nil {
+		f.Close()
+		return nil, err
+	}
+	r := new(RevReader)
+	err = r.init(f, fi.Size())
+	if err != nil {
+		f.Close()
+		return nil, err
+	}
+	return r, nil
+}
+
+// NewRevReader ...
+func NewRevReader(r io.ReaderAt, size int64) (*RevReader, error) {
+	if size < 0 {
+		return nil, errors.New("rev: size cannot be negative")
+	}
+	rr := new(RevReader)
+	err := rr.init(r, size)
+	if err != nil {
+		return nil, err
+	}
+	return rr, nil
+}
+
+func (rr *RevReader) init(r io.ReaderAt, size int64) error {
```

Comment:
I think it would be better to:

1.  Change from a method to a function that accepts a variable of type RevReader as an argument.

2. Remove size as an argument (since you'll only be reading one block).

3. Add offset (where the block is inside a file) as an argument.

4. Add filename (which rev*.dat file the block is in) as an argument.